library/net.cpp

```diff
@@ -1410,3 +1415,279 @@ int dnet_trans_forward(struct dnet_io_req *r, struct dnet_net_state *orig, struc
 
 	return dnet_trans_send(t, r);
 }
+
+// Helper for binding noncopyable to std::function
+template <class TNoncopyable>
+std::function<void ()> n2_bind_noncopyable(std::function<void (TNoncopyable)> func,
+                                           TNoncopyable param) {
+	return [func = std::move(func), param = std::make_shared<TNoncopyable>(std::move(param))]() mutable {
+		func(std::move(*param));
+	};
+}
+
+dnet_cmd n2_convert_to_response_cmd(dnet_cmd cmd) {
+	cmd.flags = (cmd.flags & ~(DNET_FLAGS_NEED_ACK)) | DNET_FLAGS_REPLY;
+	return cmd;
+}
+
+n2_repliers n2_make_repliers_via_request_queue(dnet_net_state *st, const dnet_cmd &cmd, n2_repliers repliers) {
+	auto enqueue_response = [st, cmd = n2_convert_to_response_cmd(cmd)](std::function<void ()> response_holder){
+		auto r = static_cast<dnet_io_req *>(calloc(1, sizeof(dnet_io_req)));
+		if (!r)
+			throw std::bad_alloc();
+
+		r->io_req_type = DNET_IO_REQ_TYPED_RESPONSE;
+		r->response_info = new n2_response_info({ cmd, std::move(response_holder) });
```

Comment:
May be it's not so matter, but if `new` will throw `std::bad_alloc`, memory for `r` will not be freed.

Reply:
And in `old_protocol_mounter_t::old_protocol_mounter_t()` the same.

Reply:
Here you're right, I have a bug. In `old_protocol_mounter_t()` it's ok.